react-app/src/app/components/Topbar/index.js

```diff
@@ -1,39 +1,18 @@
 import React, { Component } from 'react';
-import { NavLink } from 'react-router-dom';
 import PropTypes from 'prop-types';
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import Navbar from '../Navbar';
 
 import style from './styles.scss';
 
 class Topbar extends Component {
-  renderLink = (link, i) => {
-    if ((link.restrict && this.props.logged) || !link.restrict) {
-      return (
-        <NavLink
-          key={i.toString()}
-          to={link.to}
-          exact
-          activeClassName={style.topbarActive}
-          className={style.topbarLink}
-          onClick={link.isLogout && this.props.logout}
-        >
-          <FontAwesomeIcon icon={link.icon} />
-          &nbsp;
-          {link.name}
-        </NavLink>
-      );
-    }
-    return null;
-  };
-
   render() {
-    const links = this.props.links.map(this.renderLink);
     return (
       <header className={this.props.className}>
         <div className={style.topbar}>
           <img src={this.props.logo} className={style.topbarLogo} alt="logo" />
           <h1 className={style.topbarTitle}>{this.props.title}</h1>
-          <nav className={style.topbarNav}>{links}</nav>
+          <Navbar links={this.props.links} logged={this.props.logged} />
```

Comment:
I wouldnt create a Navbar component, since this Topbar component is enough. I would create a `Link` component that receives a link and an index, and returns the NavLink. Then you would do `this.props.links.map((link, i) => <NavLink link={link} i={i} />)`

Reply:
I think that the Navbar composes if necessary because it allows to use the same representation of links in the topbar and in the user's links (dashboard, history, points). On the other hand I agree to create the NavLink component.

Reply:
I meant component instead of composes